src/Compiler/Interactive/FSharpInteractiveServer.fs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+namespace FSharp.Compiler.Interactive
+
+open System
+open System.Text
+open System.Diagnostics
+open System.IO
+open System.IO.Pipes
+open System.Threading
+
+module CtrlBreakHandlers =
+
+    let interuptCommand = "Interactive-CtrlCNotificationCommand-Interupt"
+
+    let lineInteruptCommand =
+        Encoding.UTF8.GetBytes(interuptCommand + Environment.NewLine)
+
+    let connectionTimeout = 1000
+
+    [<AbstractClass>]
+    type public CtrlBreakService(channelName: string) =
+
+        abstract Interrupt: unit -> unit
+
+        // Exceptions percolate to callsite, IO exceptions must be handled by caller
+        // Should be run on a new thread
+        member this.Run() : unit =
+            let service = Some(new NamedPipeServerStream(channelName, PipeDirection.In))
```

Comment:
There's probably no need to wrap this in an option here.

Reply:
true